pkgs/test_core/lib/src/runner.dart

```diff
@@ -74,9 +75,24 @@ class Runner {
         var engine =
             Engine(concurrency: config.concurrency, coverage: config.coverage);
 
-        var reporterDetails = allReporters[config.reporter];
+        Reporter createFileReporter(String reporterName, String filepath) {
+          final sink = (File(config.fileReporters[reporterName])
+                ..createSync(recursive: true))
+              .openWrite();
+          return allReporters[reporterName]
+              .factory(config, engine, sink, onDone: sink.flush);
```

Comment:
How about adding a `List<IOSink>` field on this class, and flushing/closing them around here https://github.com/dart-lang/test/blob/eb8aa6b518860efc105e446b66334f531c8118ba/pkgs/test_core/lib/src/runner.dart#L233

Reply:
Yeah that is much cleaner 👍